src/pint/toa.py

```diff
@@ -1951,6 +1968,7 @@ def write_TOA_file(
         commentflag=None,
         order_by_index=True,
         *,
+        include_pn=True,
```

Comment:
I think it's easier to just `remove_pulse_numbers` before saving rather than add yet another option to `write_TOA_file`.